### PR TITLE
Fixed #169: update var type to hold bigger w&h for ellipse

### LINKED ISSUE #169: gdImageEllipse() has overflow issue for moderately large radii
On architectures where `long`s are signed 32bit values (e.g. LLP64), moderately large values of w and h (e.g. w=h=2200) cause gdImageEllipse() to overflow when [assigning to rx](https://github.com/libgd/libgd/blob/4e701317ff304ab6cb71c9f6c822944fd81bf965/src/gd.c#L1891), what results in broken ellipses. It appears to be sufficient to declare r,rx and ry as `long long`.

See the [bug reported downstream for PHP](https://bugs.php.net/bug.php?id=51498) for some more details.

### PR DESCRIPTION
**Pull request by willson-chen**

### CODE REVIEW

**Comment by willson-chen** [comment]
`int64_t` is defined in `stdint.h`

Reference to #169. In function `gdImageEllipse` and `gdImageFilledEllipse`, `r` should be signed. So I use `int64_t` instead of `uint64_t`.

**Comment by vapier** [comment]
just assume `int64_t` and related types are available.  if we want glue code, we can do it in headers, not inline code.

**Comment by willson-chen** [comment]
> just assume `int64_t` and related types are available. if we want glue code, we can do it in headers, not inline code.

could you give me more hint, please?

 We include `config.h` in `gd.c`, to see if `HAVE_STDINT` is defined. And then we can use `int64_t`. If we want to define a new type in header, such as `gd.h`, and it depends on `stdint.h`, we include `config.h` in `gd.h`?

**Comment by vapier** [comment]
i'm saying use `int64_t` unconditionally here.  wait for someone to complain and we can figure out a header replacement if/when it happens.  these standards have been around for so long that it's not worth our time imo to splatter #ifdef throughout the codebase.

**Comment by willson-chen** [comment]
> i'm saying use `int64_t` unconditionally here. wait for someone to complain and we can figure out a header replacement if/when it happens. these standards have been around for so long that it's not worth our time imo to splatter #ifdef throughout the codebase.

get it

**Comment by willson-chen** [comment]
@vapier hi, could you please review again and offer more suggestion?

**Comment by willson-chen** [comment]
> Thanks for the PR! Basically, this looks good to me, but I would prefer to check for valid inputs to these functions (e.g. what happens if `w < 0` and/or `h < 0`?), and to let the functions fail early (i.e. by returning `GD_FALSE` before drawing any pixels; changing the return type would be a BC break, though). This would require a thorough range analysis for all values, but if that is done, we likely could choose suitable types for the local variables.
> 
> Also, having (regression) tests appears to be reasonable.

I will add a test case. Since changing return type will cause BC break, shall we leave it to GD3? Or GD2.3 is fined?

**Comment by willson-chen** [comment]
@cmb69 hi, what about improving this PR by just adding test case without modifying the return type to keep API compatible. But I will leave an issue to remind us change it in GD3, and it's better to add it to milestone 3.0.

**Comment by cmb69** [comment]
Adding a return type is better left for GD 3.0. The current PR plus a test case would be fine for me. Thanks.

**Comment by willson-chen** [comment]
Test will failed in 32-bit platform in current master branch of GD, and will success with this patch.
test_master

File: tests/gdimageellipse/bug00169.c [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+int main()
+{
+	gdImagePtr im;
+	int white = 0;
+
+	im = gdImageCreateTrueColor(2200, 2200);
+	white = gdImageColorAllocate(im, 255, 255, 255);
+
+	gdImageEllipse(im, 1100, 1100, 2200, 2200, white);
+  gdAssertImageEqualsToFile("gdimageellipse/bug00169_exp.png", im);
```
Comment (cmb69) [inline tests/gdimageellipse/bug00169.c]: Nit: indentation (also in the other test below)

Comment (willson-chen) [reply tests/gdimageellipse/bug00169.c]: Fixed!

Sorry I used a new computer with vim config auto expanding tab to spaces.

Comment (willson-chen) [reply tests/gdimageellipse/bug00169.c]: Thanks for your careful review :)

**Pull request merged**